crypto/cms/cms_sd.c

```diff
@@ -916,9 +922,16 @@ int CMS_SignerInfo_verify(CMS_SignerInfo *si)
         goto err;
     }
     mctx = si->mctx;
+    if (si->pctx != NULL) {
+        EVP_PKEY_CTX_free(si->pctx);
+        si->pctx = NULL;
+    }
```

Comment:
unlike the case in `CMS_SignerInfo_verify_content` below
`si->pctx` is usually NULL here, but I can provoke the non-null pctx
by injecting a redundant `CMS_verify`